pulp_container/app/token_verification.py

```diff
@@ -167,8 +163,32 @@ def has_permission(self, request, view):
         """
         Decide upon permission based on token
         """
-        try:
-            decoded_token = request.auth
-            return _contains_accessible_actions(decoded_token, *_access_scope(request))
-        except Exception:
+        decoded_token = request.auth
+        if decoded_token is None:
             return False
+
+        try:
+            scope = init_scope(request)
+        except HasNoScope:
+            return is_requesting_root_endpoint(decoded_token)
+        else:
+            return contains_accessible_actions(decoded_token, scope)
+
+
+def is_requesting_root_endpoint(decoded_token):
+    """
+    Returns True if no access type was detected.
+    """
+    return len(decoded_token["access"]) == 0
```

Comment:
I don't see the value in this extra function. You can just insert the expression in the place where you are using it.
And since it's purpose is not so obvious, that seems like the right place for one line of comment (the function name would do just well).

Reply:
Alternatively, one can just name a variable that way.
Comments tend to get out of date, naming a boolean condition makes it self-explanatory but also doesn't require creating a function.

```
is_requesting_root_endpoint = len(decoded_token["access"]) == 0
```